src/test/scala/com/cognite/sdk/scala/common/DataPointsResourceBehaviors.scala

```diff
@@ -41,13 +41,13 @@ trait DataPointsResourceBehaviors extends Matchers with OptionValues with RetryW
           p => p.datapoints should have size 3
         )
 
-        retryWithExpectedResult[Option[DataPoint]](
```

Comment:
Those have been known to be sensitive to delays, so I don't think `retryWithExpectedResults` should be removed. It will make those tests flaky again.

Reply:
I agree, since `testDataPoints` is generated with `startTime = now` and `endTime = startTime + 20*1000`, with the sensitive to delays as you said and the default value of `before` = "now", we have to use the retry so I will keep these tests. But I also add new tests, which specify `start.plusMillis(1)` and  `end.plusMillis(1)` as the value for `before`, in this case it has to retrieve the first and last dataPoint of `testDataPoints`, what do you think?

Reply:
> 
> It actually isn't the same - the same would be if we also use `"now"` as the default argument.
> 
> Using `Instant.now()` uses the clock on the client, while `"now"` uses the clock on the server. The difference can lead to subtle bugs and what would appear as issues with consistency.
> 
> We can add `getLatestDataPointsInstant` to support passing an instant if that's useful, but the default should mirror the API spec and have "before" as a string, in my opinion.

Right, good catch 👍 , I update the code  using "before" as a string.
FYI @hakontro , @exyi  to fix the [bug reported by Ana on slack](https://cognitedata.slack.com/archives/C93QJMZE0/p1635859815160300), we will have to update [the call in spark data source](https://github.com/cognitedata/cdp-spark-datasource/blob/5557b9214e58f6e8e954766aa229d43ca23e03f9/src/main/scala/cognite/spark/v1/NumericDataPointsRdd.scala#L252) as below
`client.dataPoints.getLatestDataPoints(ids, ignoreUnknownIds = true, end.toEpochMilli.toString)` , makes sense?

Reply:
Sounds good. I think the eventual consistency of data points will be gone fairly soon (once GCP clusters have switched to the new backend), but for now, we do need the retries.